lib/run_part.c

```diff
@@ -15,23 +15,23 @@
 #include "shadowlog_internal.h"
 
 
-int run_part (char *script_path, const char *name, const char *action)
+static int run_part (char *script_path, const char *name, const char *action)
 {
-	int pid;
+	pid_t pid;
 	int wait_status;
-	int pid_status;
+	pid_t pid_status;
 	char *args[] = { script_path, NULL };
 
 	pid=fork();
 	if (pid==-1) {
-		perror ("Could not fork");
+		fprintf (shadow_logfd, "Could not fork: %s\n", strerror(errno));
 		return 1;
 	}
 	if (pid==0) {
 		setenv ("ACTION",action,1);
 		setenv ("SUBJECT",name,1);
 		execv (script_path,args);
-		perror ("execv");
+		fprintf (shadow_logfd, "execv: %s\n", strerror(errno));
```

Comment:
While changing the perror after fork, it might be good to change the string to "fork: %s\n", just like the others, for consistency.